f8a_report/v2/report_generator.py

```diff
@@ -0,0 +1,488 @@
+                self.populate_key_count(unknown_licenses),
+            'unique_cves':
+                self.populate_key_count(all_cve_list),
+            'total_average_response_time':
+                '{} ms'.format(total_response_time['all'] / len(report_content['stacks_details'])),
+            'cve_report': CVE().generate_cve_report(updated_on=start_date)
+        }
+
+    def set_average_response_time(self) -> None:
+        """Set Average Response time in self."""
+        logger.info("Calculating Average response time.")
+        for ecosytem in ('npm', 'maven', 'pypi'):
+            if self.total_stack_requests[ecosytem] > 0:
+                self.avg_response_time[ecosytem] = \
+                    self.total_response_time[ecosytem] / self.total_stack_requests[ecosytem]
+                continue
+            self.avg_response_time[ecosytem] = 0
+
+    def normalize_worker_data(self, stacks_data, retrain, frequency='daily'):
+        """Parser for worker data for Stack Analyses v2.
+
+        :arg:
+            stacks_data: Stacks Collected from DB within time-frame.
+            frequency: Frequency of Report ( daily/monthly )
+        :return: Final Venus Report Generated.
+        """
+        logger.info("Normalising v2 Stack Data.")
+        stacks_data = json.loads(stacks_data)
+        report_name = self.get_report_name(frequency, self.end_date)
+        report_template = self.get_report_template(self.start_date, self.end_date)
+
+        report_content = self.analyse_stack(stacks_data, report_template)
+
+        self.unique_stacks_with_recurrence_count = {
+            'npm': self.populate_key_count(self.stacks_list['npm']),
+            'maven': self.populate_key_count(self.stacks_list['maven']),
+            'pypi': self.populate_key_count(self.stacks_list['pypi'])
+        }
+        self.unique_stacks_with_deps_count = \
+            self.set_unique_stack_deps_count(self.unique_stacks_with_recurrence_count)
+        self.set_average_response_time()
+
+        # Get a list of unknown licenses
+        for lic_dict in self.flatten_list(self.all_unknown_lic):
+            if 'license' in lic_dict:
+                self.unknown_licenses.append(lic_dict['license'])
+
+        unknown_deps_ingestion_report = UnknownDepsReportHelper().get_current_ingestion_status()
+
+        report_content['stacks_summary'] = self.build_report_summary(
+            unknown_deps_ingestion_report, report_content)
+
+        if frequency == 'monthly':
+            # monthly data collection on the 1st of every month
+            self.collate_raw_data(self.unique_stacks_with_recurrence_count, frequency)
+
+        if retrain:
+            return self.unique_stacks_with_recurrence_count
+
+        venus_input = [frequency, report_name, report_content]
+        logger.info("Venus Report Successfully Generated.")
+        return venus_input
+
+    def get_report(self, start_date, end_date, frequency='daily', retrain=False):
```

Comment:
Cam this method be split into 2 , one for the ingestion report and the other for the stacks report.

Reply:
As per previous comment, if we dont need ingestion reporting in v2.. does this comment stands valid..?

Reply:
this will be the class that drives everything. In case of ingestion it can go to reporthelper, in case of stack it can come here.

Reply:
i am afraid, we can't do that. Ingestion is not Pluggable in v1

https://github.com/fabric8-analytics/f8a-stacks-report/blob/b028168767f4afdf24f4215653bb7d748da517f4/f8a_report/report_helper.py#L819

If we anyway want to trigger ingestion, we have to trigger whole new function.

Reply:
Ack! As discussed, we will not be having Ingestion and Sentry reporting functionality in v2.
Removing all references to Ingestion and Sentry from v2.